cfme/infrastructure/virtual_machines.py

```diff
@@ -708,17 +751,20 @@ def migrate_vm(self, email=None, first_name=None, last_name=None,
             prov_data = cfme_data["management_systems"][self.provider.key]["provisioning"]
         except (KeyError, IndexError):
             raise ValueError("You have to specify the correct options in cfme_data.yaml")
-        provisioning_data = {
+
+        request_data = {
             'request': {
                 'email': email,
                 'first_name': first_name,
-                'last_name': last_name},
-            'environment': {"host_name": {'name': prov_data.get("host")}},
+                'last_name': last_name
+            }
         }
-        if not self.provider.one_of(RHEVMProvider):
-            provisioning_data['environment']["datastore_name"] = {
-                "name": prov_data.get("datastore")}
-        view.form.fill_with(provisioning_data, on_change=view.form.submit)
+        view.form.fill_with(request_data)
+        host = host or prov_data.get("host")
+        datastore = datastore or prov_data.get("datastore")
+        view.form.environment.host_name.row(name=host).click()
```

Comment:
I tried with fills and i have quite some problems to make it works... @mfalesni also thinks that using click would be better on tables, because they act badly with fill.

Reply:
**REQUIRED**
There is special `SelectTable` widget https://github.com/ManageIQ/integration_tests/blob/master/cfme/common/vm_views.py#L140. `fill()` must be used here.